src/platform/device.gni

```diff
@@ -47,21 +47,21 @@ declare_args() {
       chip_device_platform == "linux" || chip_device_platform == "esp32"
 }
 
-_chip_device_layer = "none"
+chip_device_layer = "none"
```

Comment:
Couldn't we just use `chip_device_platform` and keep `_chip_device_layer` private?

Reply:
Unfortunately it is not so obvious, as chip_device_platform might not be equal to none and it will not work anyway, because it may take few other not supported values like "auto", "external", "freertos" and so on. That's why I wanted to avoid creating complicated condition. Is it that wrong that chip_device_layer would not be private?

Reply:
"auto" is resolved at the beginning of device.gni and other values like "external" or "freertos" indicate some kind of device layer, so it's probably OK to assume that they can use the accessory library. My concern is that if there are two arguments which mean almost the same, but not exactly the same, people may find it confusing and may not know which one to use. Eventually some people may start using "chip_device_platform" and the others "chip_device_layer". Thus as a general rule it's better to have fewer public arguments than more.